boa/src/syntax/parser/function/mod.rs

```diff
@@ -56,37 +63,47 @@ impl<R> TokenParser<R> for FormalParameters
 where
     R: Read,
 {
-    type Output = Box<[node::FormalParameter]>;
+    type Output = FormalParameterList;
 
     fn parse(self, cursor: &mut Cursor<R>) -> Result<Self::Output, ParseError> {
         let _timer = BoaProfiler::global().start_event("FormalParameters", "Parsing");
         cursor.set_goal(InputElement::RegExp);
 
         let mut params = Vec::new();
-        let mut param_names = HashSet::new();
-
-        if cursor.peek(0)?.ok_or(ParseError::AbruptEnd)?.kind()
-            == &TokenKind::Punctuator(Punctuator::CloseParen)
-        {
-            return Ok(params.into_boxed_slice());
+        let mut is_simple = true;
+        let mut has_duplicates = false;
+
+        let next_token = cursor.peek(0)?.ok_or(ParseError::AbruptEnd)?;
+        if next_token.kind() == &TokenKind::Punctuator(Punctuator::CloseParen) {
+            return Ok(FormalParameterList {
+                parameters: params.into_boxed_slice(),
+                is_simple,
+                has_duplicates,
+            });
         }
+        let start_position = next_token.span().start();
+
+        let mut parameter_names = HashSet::new();
```

Comment:
```suggestion
        let mut parameter_names = FxHashSet::new();
```

Let's leverage that we are already using `FxHash`.